subsys/dm/dm.c

```diff
@@ -243,7 +248,7 @@ static void mpsl_nonpreemptible_thread(void)
 		}
 	}
 }
-
+#if (!IS_ENABLED(CONFIG_DM_MODULE_RPC_HOST))
 static void process_data(const nrf_dm_report_t *data)
 {
 	if (!data) {
```

Comment:
I believe you can get rid of #if directive the following way

```suggestion
	if (!IS_ENABLED(CONFIG_DM_MODULE_RPC_HOST)) {
	    (...)
	} else {
	     return 0;
	}
```